plugins/leaderboard/lib/api/filters/KalturaUserScorePropertiesFilter.php

```diff
@@ -0,0 +1,184 @@
+<?php
+/**
+ * @package plugins.leaderboard
+ * @subpackage api.filters
+ * @abstract
+ * @relatedService UserScoreService
+ */
+class KalturaUserScorePropertiesFilter extends KalturaUserScorePropertiesBaseFilter
+{
+	static private $map_between_objects = array
+	(
+		"userIdEqual" => "_eq_puser_id",
+		"userIdIn" => "_in_puser_id",
+		"rankGreaterThanOrEqual" => "_gte_rank",
+		"rankLessThanOrEqual" => "_lte_rank",
+	);
+	
+	static private $order_by_map = array
+	(
+		"+rank" => "+rank",
+		"-rank" => "-rank",
+	);
+	
+	/**
+	 * @var int
+	 */
+	public $virtualEventId;
+	
+	/**
+	 * @var int
+	 */
+	public $leaderboardId;
+	
+	/**
+	 * @var string
+	 */
+	public $userIdEqual;
+	
+	/**
+	 * @var string
+	 */
+	public $userIdIn;
+	
+	/**
+	 * @var int
+	 */
+	public $ranksAboveUser;
+	
+	/**
+	 * @var int
+	 */
+	public $ranksBelowUser;
+	
+	public function getMapBetweenObjects()
+	{
+		return array_merge(parent::getMapBetweenObjects(), self::$map_between_objects);
+	}
+	
+	protected function getCoreFilter()
+	{
+		return null;
+	}
+	
+	private function initRedisWrapper()
+	{
+		$redisWrapper = new kInfraRedisCacheWrapper();
+		$redisConfig = kConf::get('redis', kConfMapNames::RUNTIME_CONFIG);
```

Comment:
there should be dedicated redis config for the leaderboard service so the name should be more meaningful